sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -495,3 +495,20 @@ def test_S_srepr_is_identity():
     p = Piecewise((10, Eq(x, 0)), (12, True))
     q = S(srepr(p))
     assert p == q
+
+
+def test_issue_10258():
+    assert Piecewise((0, x < 1), (1, True)).is_zero is None
+    assert Piecewise((-1, x < 1), (1, True)).is_zero is False
+    a = Symbol('a', zero=True)
+    assert Piecewise((0, x < 1), (a, True)).is_zero
+    assert Piecewise((1, x < 1), (a, x < 3)).is_zero is None
+    a = Symbol('a')
+    assert Piecewise((0, x < 1), (a, True)).is_zero is None
+    assert Piecewise((0, x < 1), (1, True)).is_nonzero is None
+    assert Piecewise((1, x < 1), (2, True)).is_nonzero
+    assert Piecewise((0, x < 1), (oo, True)).is_finite is None
+    assert Piecewise((0, x < 1), (1, True)).is_finite
+    assert Piecewise(([1], x < 1)).is_finite is None
```

Comment:
I was puzzled by this when I encountered it. What object will you return so that when you substitute y=1 you get the list [-2. 2]? The syntax is solve(function) -> `list of solutions` not `[list of solutions]`, so `[-2, 2]` not `[[-2, 2]]` Piecewise allows us to say "this list under this condition otherwise this list."